srtcore/core.cpp

```diff
@@ -6805,10 +6805,25 @@ int CUDT::sendmsg2(const char *data, int len, SRT_MSGCTRL& w_mctrl)
 
         // Now seqno is the sequence to which it was scheduled
         // XXX Conversion from w_mctrl.srctime -> steady_clock::time_point need not be accurrate.
-        HLOGC(dlog.Debug, log << CONID() << "buf:SENDING (BEFORE) srctime:" << FormatTime(ts_srctime)
+        HLOGC(dlog.Debug, log << CONID() << "buf:SENDING (BEFORE) srctime:"
+                << (w_mctrl.srctime ? FormatTime(ts_srctime) : "none")
                 << " DATA SIZE: " << size << " sched-SEQUENCE: " << seqno
                 << " STAMP: " << BufferStamp(data, size));
 
+        if (w_mctrl.srctime && w_mctrl.srctime < count_microseconds(m_stats.tsStartTime.time_since_epoch()))
+        {
+            LOGC(mglog.Warn,
+                log << "Wrong source time was provided. Sending is rejected.");
+            throw CUDTException(MJ_NOTSUP, MN_INVALMSGAPI);
+        }
+
+        if (w_mctrl.srctime && (!m_bMessageAPI || !m_bTsbPd))
+        {
+            LOGC(mglog.Warn,
```

Comment:
This log will spread lots of logs into the logger with every single packet (besides, it should be `dlog`, not `mglog`). Maybe only for a few packets in the beginning - but only as long as you state that times are negative only in the beginning, while in general you can't be certain of that either.

Adding this message once, then, requires an extra boolean field so that it is displayed just once (you can bet that if this time is once specified, it will be done so with every packet during the whole connection). I think it's simply better off without a message; I know it's tempting, but this log is heavy by definition.

Reply:
Changed to heavy log.